lib/rubocop/cop/chef/correctness/block_guard_clause_string_only.rb

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright:: Copyright 2019, Chef Software Inc.
+# Author:: Tim Smith (<tsmith@chef.io>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module RuboCop
+  module Cop
+    module Chef
+      # Use secure Github and Gitlab URLs for source_url and issues_url
+      #
+      # @example
+      #
+      #   # bad
+      #   template '/etc/foo' do
+      #     mode '0644'
+      #     source 'foo.erb'
+      #     only_if { 'test -f /etc/foo' }
+      #   end
+      #
+      #   # good
+      #   template '/etc/foo' do
+      #     mode '0644'
+      #     source 'foo.erb'
+      #     only_if 'test -f /etc/foo'
+      #   end
+      #
+      class BlockGuardWithOnlyString < Cop
+        MSG = 'A resource guard (not_if/only_if) that is a string should not be wrapped in {}. Wrapping a guard string in {} causes it be executed as Ruby code which will always returns true instead of a shell command that will actually run.'.freeze
+
+        def_node_matcher :block_guard_with_only_string?, <<-PATTERN
+          (block (send nil? ${:not_if :only_if}) (args) (str $_) )
```

Comment:
`nil?` might actually be overly strict since you can `resource.only_if { "something" }` as well and that might happen in some kind of dynamic building of a resource / accumulator thingy.

(at least assuming i understand how `nil?` and `_` work...)